app/services/workflow_executions/preparation_service.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+require 'csv'
+require 'tempfile'
+
+module WorkflowExecutions
+  # Service used to Prepare a WorkflowExecution
+  class PreparationService < BaseService
+    def initialize(workflow_execution, user = nil, params = {})
+      super(user, params)
+      @workflow_execution = workflow_execution
+      @samplesheet_headers = %w[sample fastq_1 fastq_2]
+      @samplesheet_rows = []
+    end
+
+    def execute # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
+      # confirm params/permissions
+      # build workflow execution run directory
+      generate_run_dir
+
+      # does it make sense to rename the SamplesWorkflowExecution model to something better
+      # e.g. SampleWorkflowExecutionParams
+      @workflow_execution.samples_workflow_executions.each do |sample_workflow_execution|
+        attachments = parse_attachments_from_samplesheet(sample_workflow_execution.samplesheet_params)
+        samplesheet_params = sample_workflow_execution.samplesheet_params
+
+        attachments.each do |key, attachment|
+          samplesheet_params[key] = copy_attachment_to_run_dir(attachment, sample_workflow_execution)
+        end
+
+        @samplesheet_rows << samplesheet_params
+      end
+
+      # persist samplesheet in run dir
+      @workflow_execution.inputs.attach(io: samplesheet_file, key: generate_input_key('samplesheet.csv'),
```

Comment:
Since we are creating and managing a new csv file `samplesheet.csv` which contains the sample sheet data we should name it such that it won't accidentally be confused with the Illumina formatted csv for fastq conversion `SampleSheet.csv`.

I'd say name it something like `samplesheet_params.csv` or `samplesheet_data.csv`

Reply:
This file is just created to run the pipeline in the `CompletionService` we will remove it, it just temporary and we won't be storing Illumina samplesheets so I don't see any chance of confusion.